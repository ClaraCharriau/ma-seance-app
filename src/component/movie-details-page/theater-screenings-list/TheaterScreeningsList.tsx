import { useLoaderData } from 'react-router-dom';
import { TheaterScreenings } from '../../../models/TheaterScreenings';
import style from './TheaterScreeningsList.module.css';
import TimeSlotsList from '../../common/time-slots-list/TimeSlotsList';

const TheaterScreeningsList = () => {
    const showtimes = useLoaderData() as TheaterScreenings[];

    return (
        <section className={style.showtimesSection}>
            <h3 className={style.sectionTitle}>Dans vos cin√©mas favoris :</h3>
            {showtimes.map(showtimes => (
                <div key={showtimes.theater.id} className={style.theaterSchedule}>
                    <p className={style.theaterName}>{showtimes.theater.name}</p>
                    <TimeSlotsList schedule={showtimes.schedule} />
                </div>
            ))}
        </section>
    );
};

export default TheaterScreeningsList;
