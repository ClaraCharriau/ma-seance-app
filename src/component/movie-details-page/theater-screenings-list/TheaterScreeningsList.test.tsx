import { render, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';
import mockScreeningsData from '../../../mocks/movies/theaters-screenings-by-movie-id-and-day-1.json';
import TheaterScreeningsList from './TheaterScreeningsList';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLoaderData: () => {
        return mockScreeningsData;
    }
}));

describe('Movie Details Component', () => {
    jest.useFakeTimers().setSystemTime(new Date('2020-01-01'));
    it('renders movie details component', () => {
        // Given
        let component: any;

        // When
        act(() => {
            component = render(
                <BrowserRouter>
                    <TheaterScreeningsList />
                </BrowserRouter>
            );
        });

        // Then
        waitFor(() => {
            expect(component.container).toMatchSnapshot();
        });
    });
});
