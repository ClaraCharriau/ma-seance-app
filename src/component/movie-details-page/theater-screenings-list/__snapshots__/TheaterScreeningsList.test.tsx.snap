// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details Component renders movie details component 1`] = `
<div>
  <section
    class="showtimesSection"
  >
    <h3
      class="sectionTitle"
    >
      Dans vos cinémas favoris :
    </h3>
    <div
      class="theaterSchedule"
    >
      <p
        class="theaterName"
      >
        UGC Ciné Cité Strasbourg Etoile
      </p>
      <div
        class="timeSlotWrapper"
      >
        <button
          class="timeSlot"
        >
          11:45
        </button>
        <button
          class="timeSlot"
        >
          14:15
        </button>
        <button
          class="timeSlot"
        >
          16:45
        </button>
        <button
          class="timeSlot"
        >
          19:15
        </button>
        <button
          class="timeSlot"
        >
          21:45
        </button>
      </div>
    </div>
    <div
      class="theaterSchedule"
    >
      <p
        class="theaterName"
      >
        Star Saint-Exupery
      </p>
      <div
        class="timeSlotWrapper"
      >
        <button
          class="timeSlot"
        >
          11:45
        </button>
        <button
          class="timeSlot"
        >
          14:15
        </button>
        <button
          class="timeSlot"
        >
          16:45
        </button>
        <button
          class="timeSlot"
        >
          19:15
        </button>
        <button
          class="timeSlot"
        >
          21:45
        </button>
      </div>
    </div>
    <div
      class="theaterSchedule"
    >
      <p
        class="theaterName"
      >
        Cinéma Vox
      </p>
      <div
        class="timeSlotWrapper"
      >
        <button
          class="timeSlot"
        >
          11:45
        </button>
        <button
          class="timeSlot"
        >
          14:15
        </button>
        <button
          class="timeSlot"
        >
          16:45
        </button>
        <button
          class="timeSlot"
        >
          19:15
        </button>
        <button
          class="timeSlot"
        >
          21:45
        </button>
      </div>
    </div>
  </section>
</div>
`;
